automation/pxf_regress/main.go

```diff
@@ -0,0 +1,427 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+)
+
+// Options settable from command line
+var testDir string = "."
+var initFile string
+
+// gpdiff.pl options for comparing actual results file with expected
+//
+// Unlike postgres/pg_regress, PXF includes
+//
+//	-w ignore all white space
+//	-B ignore changes lines are all blank
+//
+// TODO: rather than match/sub DETAIL (GP5) for CONTEXT (see global_init_file), should we add "-I DETAIL:" and "-I CONTEXT:"
+// TODO: rather than having to add start_ignore/end_ignore, should we add "-I HINT:"
+var baseDiffOpts []string = []string{"-w", "-B", "-I", "NOTICE:", "-I", "GP_IGNORE", "-gpd_ignore_headers", "-U3"}
+
+// Wrap os.Stdout in bufio.Writer so that we can flush stdout when writing
+// status messages.
+var stdout *bufio.Writer
+var progName string
+
+// internal variables
+var gpdiffProg string
+
+// summary file of all diffs
+var regressionDiffsName string
+var regressionDiffs *os.File
+
+var regressionSummaryName string
+var regressionSummary *os.File
+
+var successCount int = 0
+var failCount int = 0
+
+func init() {
+	stdout = bufio.NewWriter(os.Stdout)
+}
+
+func main() {
+	pathName, err := os.Executable()
+	if err != nil {
+		fatalf(err.Error())
+	}
+	progName = pathName
+
+	if len(os.Args) < 2 {
+		fatalf("%s: missing required argument: test-directory", progName)
+	}
+
+	f, err := findOtherFile("gpdiff.pl", true)
+	if err != nil {
+		fatalf("%s: cannot find gpdiff.pl: %s", progName, err.Error())
+	}
+	gpdiffProg = f
+
+	f, err = findOtherFile("global_init_file", false)
+	if err != nil {
+		fatalf("%s: cannot find 'global_init_file': %s", progName, err.Error())
+	}
+	initFile = f
+
+	testDir = os.Args[1]
+	tests, err := listTestQueries(testDir)
+	if err != nil {
+		fatalf("%s: cannot list test queries in %s: %s", progName, testDir, err.Error())
+	}
+
+	openResultFiles()
+	initializeEnvironment()
+
+	// Ready to run the tests
+	header("running regression queries")
+
+	for _, t := range tests {
+		runSingleTest(t)
+	}
+
+	if regressionSummary != nil {
+		regressionSummary.Close()
+	}
+
+	// Emit nice-looking summary message
+	printSummary(stdout, successCount, failCount)
+
+	if failCount > 0 {
+		fmt.Printf("The differences that caused some tests to fail can be viewed in the\nfile %q. A copy of\nthe test summary that you see above is saved in the file\n%q.\n\n", regressionDiffsName, regressionSummaryName)
+		os.Exit(1)
+	}
+
+	os.Remove(regressionDiffsName)
+	os.Remove(regressionSummaryName)
```

Comment:
```suggestion
	deleteResultsFiles()
```
can we move this to a helper function for symmetry with `openResultFiles()`